src/uploader.js

```diff
@@ -126,6 +129,12 @@ function uiUploader($log) {
         // Append file data:
         formData.append(key, file, file.name);
 
+        if (headers.length) {
```

Comment:
This won't work as objects don't have a length.

The following code worked for me:

if (headers) {
            for (var header_key in headers) {
                if (headers.hasOwnProperty(header_key)) {
                    xhr.setRequestHeader(header_key, headers[header_key]);
                }
            }
        }